display_sdl: Fix incorrect blend mode during display_read

Finomnis opened the pull request:
The blend mode was set in a way that multiplied pixels with the alpha value during read, which caused tests to fail.

Fixes #88714 partially, but not entirely.

Finomnis committed:
display_sdl: Fix incorrect blend mode during display_read

The blend mode was set in a way that multiplied pixels with the alpha
value during read, which caused tests to fail.

Signed-off-by: Martin Stumpf <finomnis@gmail.com>

faxe1008 commented:
Checked out your fix and run:
```
west build -b native_sim/native/64 -d build tests/drivers/display/display_read_write -- -DCONFIG_SDL_DISPLAY_DEFAULT_PIXEL_FORMAT_MONO10=y -DCONFIG_SDL_DISPLAY_USE_HARDWARE_ACCELERATOR=n -DCONFIG_ASSERT=y
```

yields:
```
Running TESTSUITE display_read_write
===================================================================
START - test_clear
 PASS - test_clear in 0.000 seconds
===================================================================
START - test_read_does_not_clear_existing_buffer

    Assertion failed at WEST_TOPDIR/zephyr/tests/drivers/display/display_read_write/src/main.c:62: verify_bytes_of_area: data not equal to disp_buffer

 FAIL - test_read_does_not_clear_existing_buffer in 0.000 seconds
===================================================================
START - test_write_to_buffer_head

    Assertion failed at WEST_TOPDIR/zephyr/tests/drivers/display/display_read_write/src/main.c:62: verify_bytes_of_area: data not equal to disp_buffer

 FAIL - test_write_to_buffer_head in 0.000 seconds
===================================================================
START - test_write_to_buffer_tail

    Assertion failed at WEST_TOPDIR/zephyr/tests/drivers/display/display_read_write/src/main.c:206: display_read_write_test_write_to_buffer_tail: data not equal to disp_buffer + (display_width * display_height / 8 - buf_size)

 FAIL - test_write_to_buffer_tail in 0.000 seconds
===================================================================
TESTSUITE display_read_write failed.
```

I do not get the ASSERT that you were experiencing?

Finomnis commented:
Hm. I didn't run it like that, I ran `west twister -p native_sim/native/64 -T tests/drivers/display/display_read_write/`.

Before that, I edited `tests/drivers/display/display_read_write/testcase.yaml` and removed `harness: display`. Then I replaced all `- native_sim` with `- native_sim/native/64`

faxe1008 commented:
Ok now I get the same error. It is caused by the fact that the height is set to 1 here:
https://github.com/zephyrproject-rtos/zephyr/blob/main/tests/drivers/display/display_read_write/src/main.c#L226

But I am wondering if that `__ASSERT` makes sense in the first place?

soburi commented:
I tested it and confirmed that the tests passed after applying this patch and reverting https://github.com/zephyrproject-rtos/zephyr/pull/88126.
I think it would be better to accept this PR for now and file a separate issue for htiled. What do you think?

soburi reviewed (approved):
@faxe1008 @Finomnis 

I tested it and confirmed that the tests passed after applying this patch and reverting https://github.com/zephyrproject-rtos/zephyr/pull/88126.
I think it would be better to accept this PR for now and file a separate issue for htiled. What do you think?

Finomnis commented:
> Yah, I'm quite sure that the tests were not adjusted after htiling was introduced.

I might have to correct that statement; it's probably another bug in the SDL driver and the tests are fine.

faxe1008 commented:
What also should be checked why the test were not run on the breaking PR

Finomnis commented:
They require a display, can they be run on a headless CI?

soburi commented:
> They require a display, can they be run on a headless CI?

Yes. It cannot run with the Docker image used for CI.

Finomnis commented:
https://discourse.libsdl.org/t/possible-to-run-sdl2-headless/25665/2

It seems SDL2 can run on headless if initialized correctly.

kartben commented:
@Finomnis it needs a second approval -- maybe from @VynDragon or @faxe1008?

Finomnis commented:
Bump
